auth/sasl/authenticatee_test.go

```diff
@@ -0,0 +1,98 @@
+package sasl
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mesos/mesos-go/auth/callback"
+	"github.com/mesos/mesos-go/auth/sasl/mech/crammd5"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/messenger"
+	"github.com/mesos/mesos-go/upid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/net/context"
+)
+
+type MockTransport struct {
+	*messenger.MockedMessenger
+}
+
+func (m *MockTransport) Send(ctx context.Context, upid *upid.UPID, msg proto.Message) error {
+	return m.Called(mock.Anything, upid, msg).Error(0)
+}
+
+func TestAuthticatee_validLogin(t *testing.T) {
+	assert := assert.New(t)
+	ctx := context.TODO()
+	client := upid.UPID{
+		ID:   "someFramework",
+		Host: "b.net",
+		Port: "789",
+	}
+	server := upid.UPID{
+		ID:   "serv",
+		Host: "a.com",
+		Port: "123",
+	}
+	tpid := upid.UPID{
+		ID:   "sasl_transport",
+		Host: "g.org",
+		Port: "456",
+	}
+	handler := callback.HandlerFunc(func(cb ...callback.Interface) error {
+		for _, c := range cb {
+			switch c := c.(type) {
+			case *callback.Name:
+				c.Set("foo")
+			case *callback.Password:
+				c.Set([]byte("bar"))
+			case *callback.Interprocess:
+				c.Set(server, client)
+			default:
+				return &callback.Unsupported{c}
+			}
+		}
+		return nil
+	})
+	var transport *MockTransport
+	factory := transportFactoryFunc(func() messenger.Messenger {
```

Comment:
Ok I see why transportFactory was made a function adapter type that returns *messenger.Messenger*.  Is this a matter of style since it would just as easily work using the *messenger.Messenger* interface.  Or is it indirection for yet-unknown impl ?